go/private/gazelle.bzl

```diff
@@ -0,0 +1,73 @@
+# Copyright 2016 The Bazel Go Rules Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _gazelle_script_impl(ctx):
+  script_content = ''
+  args = ctx.attr.args
+  script_content += 'BASE=$(pwd)\n'
+  script_content += 'WORKSPACE=$(dirname $(readlink WORKSPACE))\n'
+  script_content += 'cd $WORKSPACE\n'
+  args = [
+      "-repo_root", "$WORKSPACE",
+      "-go_prefix", ctx.attr._go_prefix.go_prefix,
+      "-external", ctx.attr.external,
+      "-mode", ctx.attr.mode,
+  ]
+  if ctx.attr.build_tags:
+    args += ["-build_tags", ",".join(ctx.attr.build_tags)]
+  script_content += '$BASE/{0} {1} $@\n'.format(ctx.file._gazelle.short_path, " ".join(args))
+  script_file = ctx.new_file(ctx.label.name+".bash")
+  ctx.file_action(output=script_file, executable=True, content=script_content)
+  return struct(
+    files = depset([script_file]),
+    runfiles = ctx.runfiles([ctx.file._gazelle])
+  )
+
+_gazelle_script = rule(
+    _gazelle_script_impl,
+    attrs = {
+        "mode": attr.string(mandatory=True, values=["print", "fix", "diff"]),
+        "external": attr.string(mandatory=True, values=["external", "vendored"]),
+        "build_tags": attr.string_list(mandatory=True),
+        "args": attr.string_list(mandatory=True),
+        "_gazelle": attr.label(
+            default = Label("@io_bazel_rules_go//go/tools/gazelle/gazelle:gazelle"),
+            allow_files = True,
+            single_file = True,
+            executable = True,
+            cfg = "host"
+        ),
+        "_go_prefix": attr.label(default = Label(
+            "//:go_prefix",
+            relative_to_caller_repository = True,
+        )),
+    }
+)
+
+def gazelle(name, mode = "fix", external="external", build_tags=[], args = []):
```

Comment:
I tried that once before, and without sh_binary I could not make a working cross platform script that bazel run would invoke.